index.js

```diff
@@ -1,81 +1,143 @@
+            this.lampDevice = undefined;
+            this.log.error('Device not connected', e);
         }
-        return this.lampDevice
+        return this.lampDevice;
     }
 
     async listenLampState(){
-        const device = await this.getLamp()
-        device.on('powerChanged', isOn => this.switch.getCharacteristic(Characteristic.On).updateValue(isOn))
+        const device = await this.getLamp();
+        device.on('powerChanged', isOn => this.lamp.getCharacteristic(Characteristic.On).updateValue(isOn));
+	    device.on('colorChanged', color => this.lamp.getCharacteristic(Characteristic.ColorTemperature).updateValue(color));
     }
 
     async getState(callback) {
-        this.log('Get state...')
+        this.log('Get state...');
         try {
-            const device = await this.getLamp()
-            const power = await device.power()
-            callback(null, power)
+            const device = await this.getLamp();
+            const power = await device.power();
+            callback(null, power);
         } catch (e) {
-            this.log.error('Error getting state', e)
-            callback(e)
+            this.log.error('Error getting state', e);
+            callback(e);
         }
     }
 
     async setState(state, callback) {
-        this.log('Set state to', state)
+        this.log('Set state to', state);
+        try {
+            const device = await this.getLamp();
+            await device.power(state);
+            callback(null);
+        } catch (e) {
+            this.log.error('Error setting state', e);
+            callback(e);
+        }
+    }
+
+    async getBrightness(callback) {
+        this.log('Get brightness...');
         try {
-            const device = await this.getLamp()
-            await device.power(state)
-            callback(null)
+            const device = await this.getLamp();
+            const brightness = await device.brightness();
+            callback(null, brightness);
         } catch (e) {
-            this.log.error('Error setting state', e)
-            callback(e)
+            this.log.error('Error getting brightness', e);
+            callback(e);
         }
     }
+	async setBrightness(state, callback) {
+		this.log('Set brightness to', state);
+		try {
+			const device = await this.getLamp();
+			await device.brightness('' + state);
+			callback(null);
+		} catch (e) {
+			this.log.error('Error setting brightness', e);
+			callback(e);
+		}
+    }
+    
+    async getColorTemperature(callback) {
+        this.log('Get color...');
+        try {
+            const device = await this.getLamp();
+            const color = await device.color();
+            callback(null, color);
+        } catch (e) {
+            this.log.error('Error getting brightness', e);
+            callback(e);
+        }
+    }
+	async setColorTemperature(value, callback) {
+        this.log('Set color to', value);
+        value = Math.round(1000000 / value);
```

Comment:
Create a new constant instead or reuse it like `const brightness`

Reply:
Added.